src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue

```diff
@@ -259,32 +263,21 @@ export default class CorrectNameRequest extends Mixins(CommonMixin, NameRequestM
   }
 
   /**
-   * Parse and Set the Name Request date to Store.
-   * @param nr The name request data
+   * Stores the new Name Request.
+   * @param nameRequest the name request object
    */
-  parseNameRequest (nr: NrResponseIF): void {
-    const nrCorrection: NrCorrectionIF = {
-      legalType: nr.legalType,
-      nrNumber: this.nameRequestNumber,
-      legalName: this.getNrApprovedName(nr) || '',
-      expiry: nr.expirationDate,
-      status: nr.state,
-      requestType: nr.request_action_cd,
-      applicant: {
-        fullName: this.formatFullName(nr.applicants),
-        fullAddress: this.formatFullAddress(nr.applicants),
-        phoneNumber: nr.applicants.phoneNumber,
-        emailAddress: nr.applicants.emailAddress
-      } as NameRequestApplicantIF
-    }
-
-    // set the new correction NR data
-    this.setNameRequest({ ...this.getNameRequest, ...nrCorrection })
+  storeNameRequest (nameRequest: NameRequestIF): void {
+    // overwrite the existing NR data with the new NR data
+    this.setNameRequest({
+      ...this.getNameRequest,
+      ...nameRequest
+    } as any)
+    this.setNameRequestLegalName(this.getNrApprovedName(nameRequest) || null)
   }
 
   /** Inform parent the process is complete. */
-  @Emit('isSaved')
-  emitIsSaved (isSaved = false): boolean {
+  @Emit('saved')
+  private emitSaved (isSaved = false): boolean {
```

Comment:
Does it matter here if we have the `private` access modifier? I thought we were moving away from them since they're not really used.

Reply:
They're not functional _yet_, but Typescript is still evolving 😉 

No, it doesn't really matter. Mostly we're well behaved and don't reach externally into components to tweak their properties or call their methods 😆 , but I think it's still good practice as it confers extra info on how the method is used.

Reply:
Got it! Sounds good to me 😃